python/cudf/cudf/core/dataframe.py

```diff
@@ -2551,6 +2555,16 @@ def insert(self, loc, name, value, nan_as_null=None):
             name or label of column to be inserted
         value : Series or array-like
         """
+        return self._insert(
+            loc=loc,
+            name=name,
+            value=value,
+            nan_as_null=nan_as_null,
+            ignore_index=False,
+        )
+
+    @annotate("DATAFRAME__INSERT", color="green", domain="cudf_python")
+    def _insert(self, loc, name, value, nan_as_null=None, ignore_index=False):
```

Comment:
It looks like the main optimization here is that indexes should not be ignored by the public API, but you've found internal usage where we should. You mentioned trying to apply #10263 to `insert` once that's merged, so I think the default parameter here should maybe be `ignore_index=True` so that we only use the index when needed. That way switching APIs will lead to the fast behavior by default (assuming that test will easily catch cases where the code actually needs to account for the index), otherwise it's not enough to switch APIs but users would also need to specify the right parameter value.

Reply:
Sure, `ignore_index` should `True` by default. Changed it 👍